apps/els_lsp/src/els_completion_provider.erl

```diff
@@ -546,6 +552,18 @@ exported_definitions(Module, POIKind, ExportFormat) ->
       []
   end.
 
+-spec module_name(els_dt_document:item()) -> [binary()].
+module_name(#{uri := Uri}) ->
+  ModuleName = filename:basename(lists:last(filename:split(Uri)),
+  filename:extension(Uri)),
+  PurifiedName = case binary:match(ModuleName, <<"-">>) of
```

Comment:
I believe it won't be enough to match on the `-` here. Quoting the [Erlang docs](https://www.erlang.org/doc/reference_manual/data_types.html#atom):

> An atom is to be enclosed in single quotes (') if it does not begin with a lower-case letter or if it contains other characters than alphanumeric characters, underscore (_), or @.

We could port [the regexp from the Emacs Erlang](https://github.com/erlang/otp/blob/master/lib/tools/emacs/erlang.el#L601) mode to Erlang LS:

```
(defconst erlang-atom-regular-regexp
    (if erlang-regexp-modern-p
        "\\_<[[:lower:]]\\(?:\\sw\\|\\s_\\)*\\_>"
      "\\<[[:lower:]]\\(?:\\sw\\|\\s_\\)*\\>")
    "Regexp describing a regular (non-quoted) atom")
```